plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadata.java

```diff
@@ -0,0 +1,564 @@
+        {
+            loadTableCallCount = 0;
+            materializedViewDefinition = Optional.empty();
+        }
+
+        @Override
+        public List<String> listNamespaces(ConnectorSession session)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public boolean dropNamespace(ConnectorSession session, String namespace)
+        {
+            return false;
+        }
+
+        @Override
+        public Map<String, Object> loadNamespaceMetadata(ConnectorSession session, String namespace)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public Optional<TrinoPrincipal> getNamespacePrincipal(ConnectorSession session, String namespace)
+        {
+            return Optional.empty();
+        }
+
+        @Override
+        public void createNamespace(ConnectorSession session, String namespace, Map<String, Object> properties, TrinoPrincipal owner)
+        {
+        }
+
+        @Override
+        public void setNamespacePrincipal(ConnectorSession session, String namespace, TrinoPrincipal principal)
+        {
+        }
+
+        @Override
+        public void renameNamespace(ConnectorSession session, String source, String target)
+        {
+        }
+
+        @Override
+        public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> namespace)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public Transaction newCreateTableTransaction(ConnectorSession session, SchemaTableName schemaTableName, Schema schema, PartitionSpec partitionSpec, String location, Map<String, String> properties)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public boolean dropTable(ConnectorSession session, SchemaTableName schemaTableName, boolean purgeData)
+        {
+            return false;
+        }
+
+        @Override
+        public void renameTable(ConnectorSession session, SchemaTableName from, SchemaTableName to)
+        {
+        }
+
+        @Override
+        public Table loadTable(ConnectorSession session, SchemaTableName schemaTableName)
+        {
+            if (++loadTableCallCount >= 2) {
+                throw new TableNotFoundException(schemaTableName);
+            }
+
+            return TEST_ICEBERG_TABLE;
+        }
+
+        @Override
+        public void updateTableComment(ConnectorSession session, SchemaTableName schemaTableName, Optional<String> comment)
+        {
+        }
+
+        @Override
+        public String defaultTableLocation(ConnectorSession session, SchemaTableName schemaTableName)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public void setTablePrincipal(ConnectorSession session, SchemaTableName schemaTableName, TrinoPrincipal principal)
+        {
+        }
```

Comment:
`throw new UnsupportedOperationException();`

Reply:
Removed this class.

Reply:
I removed this class but I think it is still good to keep: https://github.com/trinodb/trino/pull/9050#issuecomment-911645993.